src/helpers/notifications.js

```diff
@@ -0,0 +1,21 @@
+import { notificationServices } from '../services';
+
+const { notificationSystem, getNotifications } = notificationServices;
+
+export default function notifications(io) {
+  io.on('connection', (socket) => {
+    socket.on('join', async (user) => {
+      const { id } = user;
+      socket.id = id;
+      const myNotifications = await getNotifications(socket.id);
+      socket.emit('all-notifications', myNotifications);
+    });
+    // Subscribe to new-notification event
+    notificationSystem.on('new-notification', (notification) => {
+      socket.emit('new-notification', notification);
+    });
+
+    // Disconnect event
+    socket.on('disconnect', () => {});
+  });
```

Comment:
we can use middleware instead setting the id 
https://socket.io/docs/v4/middlewares/

Reply:
Thanks for the review, Useful, Let me put a hand into it.

Reply:
Resolved